mobile/lib/features/wallet/wallet_theme.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+
+@immutable
+class WalletTheme extends ThemeExtension<WalletTheme> {
+  final Color lightning;
+  final Color onChain;
+
+  final Color borderColor;
+  final Color bgColor;
+  final Color dividerColor;
+  final ButtonStyle iconButtonStyle;
+
+  WalletTheme(
+      {this.lightning = Colors.yellow,
+        this.onChain = Colors.orange,
+        this.borderColor = Colors.grey,
+        this.bgColor = Colors.white,
+        this.dividerColor = Colors.black,
+        ColorScheme? colors,
```

Comment:
🔧 I am a bit confused by this code.

`colors` is optional, but we use it with `!` below - without a null check. Or am I missing something?
I think it needs a null-check or it has to be mandatory. 

I like the idea to pass in the swatch to derive colors from it.

Reply:
It's done this way to allow either ColorScheme or ButtonStyle to be passed. First is passed when WalletTheme is created - the ColorScheme comes directly from the main ThemeData. Second is only used for copyWith and lerp. I am not sure how to express "must have one or the other" in Dart.

Reply:
I think the easiest solution for this example would be to give `ColorScheme? colors` a default value to ensure that it is always set:

```
  WalletTheme(
      {this.lightning = Colors.yellow,
        this.onChain = Colors.orange,
        this.borderColor = Colors.grey,
        this.bgColor = Colors.white,
        this.dividerColor = Colors.black,
        ColorScheme colors = ColorScheme.fromSwatch(primarySwatch: Colors.blue),
        ButtonStyle? iconButtonStyle,}):
        iconButtonStyle = iconButtonStyle ?? IconButton.styleFrom(
          foregroundColor: colors!.onPrimary,
          backgroundColor: colors.primary,
          disabledBackgroundColor: colors.onSurface.withOpacity(0.12),
          hoverColor: colors.onPrimary.withOpacity(0.08),
          focusColor: colors.onPrimary.withOpacity(0.12),
          highlightColor: colors.onPrimary.withOpacity(0.12),
        );
```

We don't use `  ColorScheme colors` internally, so I don't see a reason not to default it. But maybe I'm missing something.

---

Alternatively, you could add an `assert!` to ensure that *either* of the values is set; here's an example where I use an `assert` in a constructor in a different context:

https://github.com/get10101/10101/blob/e038857285b25c28db66d8798a666ce7e0955c7f/mobile/lib/features/trade/trade_tabs.dart#L35

Reply:
The change you've given as-is with 
```
ColorScheme colors = ColorScheme.fromSwatch(primarySwatch: Colors.blue)
```

will not work as fromSwatch is not const, so I think the assert solution may be best